client/packages/react-app/src/pages/Projects/index.js

```diff
@@ -1,51 +1,144 @@
+  const [daiAllowance, setDaiAllowance] = useState(0)
+  const [bTokenAllowance, setBTokenAllowance] = useState(0)
         
   return (
     <>
       <Jumbotron ref={headingRef}>
         <JumbotronColumn>
-            <MainHeader>Fund Project</MainHeader>
+            <MainHeader>Discover Projects</MainHeader>
         </JumbotronColumn>
         <JumbotronColumn>
-        <Button onClick={async () => {
-            const salary = await vault.salary();
-            setSelectedProjectSalary(salary);
-            const recipient = await vault.recipient();
-            setSelectedProjectRecipient(recipient);
-            const principal = await vault.principal();
-            const startedSurplus = await vault.startedSurplus();
-            setSelectedProjectPrincipal(principal);
-            setSelectedProjectStartedSurplus(startedSurplus);
-            setProjectDetailsModalIsOpen(true);
-        }}>View</Button>
+        <CardRow>
+        <CardColumn>
+              <Logo src={hardhat} alt="hardhat"/>
+          </CardColumn>
+        <CardColumn>
+          <ProjectTitleWrapper><ProjectTitle>Hardhat</ProjectTitle>
+            <Icon src={aDaiLogo} alt="aDai"/>
+            <Icon src={aaveLogo} alt="aave"/>
+          </ProjectTitleWrapper>
+              <Info>Ethereum development environment for professionals</Info>
+              <Button onClick={async () => {
+                const salary = await vault.salary();
+                const bTokenBalance = await bToken.balanceOf(account);
+                const principal = await vault.depositorToPrincipal(account);
+                const recipientReserve = await vault.recipientReserve();
+                const valueLocked = await aDai.balanceOf(addresses.vault);
+                const daiAllowance = await dai.allowance(account, addresses.vault);
+                const bTokenAllowance = await bToken.allowance(account, addresses.vault);
+                setSelectedProjectSalary(salary);
+                setSelectedProjectBTokenBalance(bTokenBalance);
+                setSelectedProjectPrincipal(principal);
+                setSelectedProjectValueLocked(valueLocked);
+                setSelectedProjectRecipientReserve(recipientReserve);
+                setDaiAllowance(daiAllowance);
+                setBTokenAllowance(bTokenAllowance);
+                setProjectDetailsModalIsOpen(true);
+            }}>View</Button>
+          </CardColumn>
+        </CardRow>
         {projectDetailsModalIsOpen ?
               <ProjectDetailsModal 
-              recipient={selectedProjectRecipient}
-              salary={selectedProjectSalary.toString()}
-              startedSurplus={selectedProjectStartedSurplus}
-              principal={selectedProjectPrincipal.toString()}
+              bTokenBalance={ethers.utils.formatEther(selectedProjectBTokenBalance)}
+              salary={ethers.utils.formatEther(selectedProjectSalary)}
```

Comment:
I forgot how React works, but essentially `setSelectedProjectSalary ` would automatically set the value for `selectedProjectSalary`, correct?

Reply:
yep and we are passing in that value for the modal component to display